File: android/app/src/main/java/gg/strims/ppspp/rpc/TransferPeerClient.kt
package gg.strims.ppspp.rpc
import gg.strims.ppspp.proto.*

class TransferPeerClient(filepath: String) : RPCClient(filepath) {

    suspend fun announceSwarm(
        arg: TransferPeerAnnounceSwarmRequest = TransferPeerAnnounceSwarmRequest()
    ): TransferPeerAnnounceSwarmResponse =
        this.callUnary("TransferPeer/AnnounceSwarm", arg)

}

File: src/lib/api/transferPeerClient.ts
import * as pb from "../pb";
import { RPCHost } from "../rpc/host";
import { Readable as GenericReadable } from "../rpc/stream";

export default class TransferPeer {
  constructor(private readonly host: RPCHost) {}

  public announceSwarm(
    arg: pb.ITransferPeerAnnounceSwarmRequest = new pb.TransferPeerAnnounceSwarmRequest()
  ): Promise<pb.TransferPeerAnnounceSwarmResponse> {
    return this.host.expectOne(
      this.host.call("TransferPeer/AnnounceSwarm", new pb.TransferPeerAnnounceSwarmRequest(arg))
    );
  }
}
